### ISSUE TITLE
SMS does not come when registering a USA number

### ISSUE THREAD

**Issue by Whomakes** [tgalal/yowsup#3036]
SMS does not come when registering a USA number from TextNow. Registration of my number goes well. localization and language set en,US
what could be the problem?

**Comment by Whomakes** [comment]
if use real phone sms come

**Comment by Whomakes** [comment]
Config:
{
    "__version__": 1,
    "cc": "1",
    "chat_dns_domain": "fb",
    "client_static_keypair": "6DXccH11B0DAVAfQK3yQpopzYzGslvYq2Hd+0GeALUxB3oOBf8gz+Pus9m50jzmb93dqXGxHlx1YMpGd6D9MPw==",
    "expid": "GqiSxviMTTKG3KnZWuRVCw==",
    "fdid": "c2c23cfe-9ff4-4458-942c-9542e5c35da9",
    "id": "GpYZ0oGWIKzqOZALVxUU0KNGPDo=",
    "login": "12495003248",
    "mcc": "310",
    "mnc": "550",
    "phone": "12495003248",
    "pushname": "TpAaiJtFh",
    "sim_mcc": "000",
    "sim_mnc": "000"
}

**Comment by Whomakes** [comment]
i tried using different mnc but no result

**Comment by mreinar** [comment]
> SMS does not come when registering a USA number from TextNow. Registration of my number goes well. localization and language set en,US
> what could be the problem?

So, here we go..
If you use US TN numbers - its mostly confirm only within voice call, BUT 90% of USA (its important) pre-banned in WA
If you use CAN TN numbers (same pref +1, sometimes same code (sic), but different HLR) - DONT EVER USE Can/ USA locale. Try Asia locale (not joke)- change proxy, timezone AND LANGUAGE (!!) within chosen locale and have a try... Use this bug everyday within more then 5k reg - works perfect

**Comment by Whomakes** [comment]
> > SMS does not come when registering a USA number from TextNow. Registration of my number goes well. localization and language set en,US
> > what could be the problem?
> 
> So, here we go..
> If you use US TN numbers - its mostly confirm only within voice call, BUT 90% of USA (its important) pre-banned in WA
> If you use CAN TN numbers (same pref +1, sometimes same code (sic), but different HLR) - DONT EVER USE Can/ USA locale. Try Asia locale (not joke)- change proxy, timezone AND LANGUAGE (!!) within chosen locale and have a try... Use this bug everyday within more then 5k reg - works perfect

can you share settings in yowsup?

**Comment by Whomakes** [comment]
i change lg and lc in http.warequest and noise.layer, set en and US. but I don't know where to change timezone
tried to use proxy asia, no result

**Comment by mreinar** [comment]
> > > SMS does not come when registering a USA number from TextNow. Registration of my number goes well. localization and language set en,US
> > > what could be the problem?
> > 
> > 
> > So, here we go..
> > If you use US TN numbers - its mostly confirm only within voice call, BUT 90% of USA (its important) pre-banned in WA
> > If you use CAN TN numbers (same pref +1, sometimes same code (sic), but different HLR) - DONT EVER USE Can/ USA locale. Try Asia locale (not joke)- change proxy, timezone AND LANGUAGE (!!) within chosen locale and have a try... Use this bug everyday within more then 5k reg - works perfect
> 
> can you share settings in yowsup?

i use it within emulators, but pretty sure - same in yowsup
its TN specific

**Comment by Whomakes** [comment]
> > > > SMS does not come when registering a USA number from TextNow. Registration of my number goes well. localization and language set en,US
> > > > what could be the problem?
> > > 
> > > 
> > > So, here we go..
> > > If you use US TN numbers - its mostly confirm only within voice call, BUT 90% of USA (its important) pre-banned in WA
> > > If you use CAN TN numbers (same pref +1, sometimes same code (sic), but different HLR) - DONT EVER USE Can/ USA locale. Try Asia locale (not joke)- change proxy, timezone AND LANGUAGE (!!) within chosen locale and have a try... Use this bug everyday within more then 5k reg - works perfect
> > 
> > 
> > can you share settings in yowsup?
> 
> i use it within emulators, but pretty sure - same in yowsup
> its TN specific

no, the problem is in yowsup. on the emulator I have no problem

**Comment by Whatsapp-Protocol** [comment]
I found that this git can be verified by voice, and after testing it, it can be registered and used normally.
you can test it
https://github.com/lovethiscode/Gorgeous-Whatsapp
